src/uri.cr

```diff
@@ -135,49 +125,63 @@ class URI
   # uri = URI.parse "http://foo.com/posts?id=30&limit=5#time=1305298413"
   # uri.full_path # => "/posts?id=30&limit=5"
   # ```
-  def full_path
+  def full_path : String
```

Comment:
Another method can also accept an IO.

Reply:
What?

Reply:
i mean an overload, `full_path(io : IO)`
This avoids creating an intermediary string when logging for example.

Reply:
I don't think that would be wise. It would be one random accessor in `URI` accepting an IO just because of how it's implemented.

Anyway, that'd be a tale for another day.